src/test/scala/org.openeo/OpenSearchClientTest.scala

```diff
@@ -2,17 +2,113 @@ package org.openeo
+    arguments(LocalDate.parse("2018-10-08"), new java.lang.Double(2.09)),
+    arguments(LocalDate.parse("2018-11-06"), new java.lang.Double(2.10)),
+    arguments(LocalDate.parse("2018-11-21"), new java.lang.Double(2.11)),
+    arguments(LocalDate.parse("2019-05-06"), new java.lang.Double(2.12)),
+    arguments(LocalDate.parse("2019-07-08"), new java.lang.Double(2.13)),
+    arguments(LocalDate.parse("2020-02-04"), new java.lang.Double(2.14)),
+    arguments(LocalDate.parse("2021-03-30"), new java.lang.Double(3.00)),
+    arguments(LocalDate.parse("2021-06-30"), new java.lang.Double(3.01)),
+    arguments(LocalDate.parse("2022-01-25"), new java.lang.Double(4.00)),
+    arguments(LocalDate.parse("2022-12-06"), new java.lang.Double(5.09)),
+    arguments(LocalDate.parse("2018-08-31"), new java.lang.Double(99.99)), // Undocumented. Manually added
+  ))
+
+  class HttpsCache extends sun.net.www.protocol.https.Handler {
+    var enabled = false
+
+    def openConnectionSuper(url: URL): java.net.URLConnection = super.openConnection(url)
+
+    override def openConnection(url: URL): java.net.URLConnection = new java.net.HttpURLConnection(url) {
+      override def getInputStream: InputStream = {
+        if (!enabled) {
+          openConnectionSuper(url).getInputStream
+        } else {
+          val fullUrl = this.url.toString
+          val idx = fullUrl.indexOf("//")
+          var filePath = fullUrl.substring(idx + 2)
+          if (filePath.length > 255) {
+            // An individual name should be max 255 characters long. Lazy implementation caps whole file path:
+            val hash = "___" + (filePath.hashCode >>> 1).toString // TODO, parse extension?
+            filePath = filePath.substring(0, 255 - hash.length) + hash
+          }
+          val lastSlash = filePath.lastIndexOf("/")
+          val (basePath, filename) = filePath.splitAt(lastSlash + 1)
+          filePath = basePath + filename
+
+          // val cachePath = "tmp/" // Use tmp to re-download files
+          val cachePath = getClass.getResource("/org/openeo/httpsCache").getPath
+          val path = Paths.get(cachePath, filePath)
+          if (!Files.exists(path)) {
+            Files.createDirectories(Paths.get(cachePath, basePath))
+            println("Caching request: " + path)
+            val stream = openConnectionSuper(url).getInputStream
+            Files.write(path, scala.io.Source.fromInputStream(stream).mkString.getBytes)
+          } else {
+            println("Using cached request: " + path)
+          }
+          new FileInputStream(new File(path.toString))
+        }
+      }
+
+      override def connect(): Unit = {}
+
+      override def disconnect(): Unit = ???
+
+      override def usingProxy(): Boolean = ???
+    }
+  }
+
+  val httpsCache = new HttpsCache()
+  // This method can be called at most once in a given Java Virtual Machine:
+  URL.setURLStreamHandlerFactory(new URLStreamHandlerFactory() {
```

Comment:
@bossie I added a kind of proxy here to greatly speed up the tests. And to be independent on the service being down. It could use some cleaning up, but do you think this approach is acceptable?
Commiting the cached files adds 5Mb to the .git folder

Reply:
@soxofaan what do you think about using a HTTP proxy in opensearch tests?